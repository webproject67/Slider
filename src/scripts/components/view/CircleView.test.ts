import CircleView from './CircleView';

const state = {
  min: 0,
  max: 100,
  from: 0,
  fromPercent: 0,
  to: 100,
  toPercent: 100,
  step: 1,
  view: false,
  range: false,
  flag: true,
  progress: true,
  scale: true,
};

describe('snapshot', () => {
  test('renders correctly element1', () => {
    expect(new CircleView(state).getElement()).toMatchSnapshot();
  });

  test('renders correctly element2', () => {
    state.view = true;
    expect(new CircleView(state).getElement()).toMatchSnapshot();
  });

  test('renders correctly element3', () => {
    state.to = 90;
    state.range = true;
    expect(new CircleView(state).getElement()).toMatchSnapshot();
  });

  test('renders correctly element4', () => {
    state.view = false;
    expect(new CircleView(state).getElement()).toMatchSnapshot();
  });
});
