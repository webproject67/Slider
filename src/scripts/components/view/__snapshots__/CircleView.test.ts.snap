// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot renders correctly element1 1`] = `
<div
  class="slider__circles"
>
  <div
    class="slider__circle slider__circle_position_maximum"
    style="left: 100%;"
  />
  <div
    class=""
  />
</div>
`;

exports[`snapshot renders correctly element2 1`] = `
<div
  class="slider__circles"
>
  <div
    class="slider__circle slider__circle_position_vertical-maximum"
    style="top: 100%;"
  />
  <div
    class=""
  />
</div>
`;

exports[`snapshot renders correctly element3 1`] = `
<div
  class="slider__circles"
>
  <div
    class="slider__circle slider__circle_position_vertical-minimum"
    style="top: 0%;"
  />
  <div
    class="slider__circle slider__circle_position_vertical-maximum"
    style="top: 100%;"
  />
</div>
`;

exports[`snapshot renders correctly element4 1`] = `
<div
  class="slider__circles"
>
  <div
    class="slider__circle slider__circle_position_minimum"
    style="left: 0%;"
  />
  <div
    class="slider__circle slider__circle_position_maximum"
    style="left: 100%;"
  />
</div>
`;
